crates/html-macro/src/parser.rs

```diff
@@ -265,20 +270,15 @@ impl HtmlParser {
                 };
 
                 if parent_to_children_indices.len() > 0 {
-                    let create_children_vec = quote! {
-                        #parent_name.children = Some(vec![]);
-                    };
-
-                    tokens.push(create_children_vec);
-
                     for child_idx in parent_to_children_indices.iter() {
                         let children =
                             Ident::new(format!("node_{}", child_idx).as_str(), Span::call_site());
 
-                        // TODO: Multiple .as_mut().unwrap() of children. Let's just do this once.
                         let push_children = quote! {
-                            for child in #children.into_iter() {
-                                #parent_name.children.as_mut().unwrap().push(child);
+                            if let VirtualNode::Element(ref mut element_node) = #parent_name {
+                                element_node.children.extend(#children.into_iter());
+                            } else {
+                                // TODO: Can this happen? What to do?
```

Comment:
Hopefully not haha... Maybe a quote_spanned! unreachable!() ..?

```rust
// Right above push children
let unreachable = quote_spanned!(Span::call_site()=>{unreachable!("Not elements cannot have children")});

// ...

// Replace the TODO with #unreachable
```

Reply:
Thanks! Done.